Sources/StreamChat/WebSocketClient/EventMiddlewares/ThreadUpdaterMiddleware.swift

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+struct ThreadUpdaterMiddleware: EventMiddleware {
+    func handle(event: Event, session: DatabaseSession) -> Event? {
+        switch event {
+        case let event as MessageReadEventDTO:
+            if let threadDetails = event.payload.threadDetails {
+                session.markThreadAsRead(parentMessageId: threadDetails.parentMessageId, userId: event.user.id, at: event.createdAt)
+            }
+        case let event as NotificationMarkUnreadEventDTO:
+            // At the moment, this event does not return the thread id, so
```

Comment:
hm, but how do you distinguish it from the regular mark unread event for messages outside threads?

Reply:
with with the lastReadMessageId, if there is non, it means it is related to threads,but ideally the backend should update this.